src/app/modules/bookmark/component/bookmark-settings/bookmark-settings.component.ts

```diff
@@ -29,6 +31,19 @@ export class BookmarkSettingsComponent implements UserSettingsComponent {
     this.addBookmark()
   }
 
+  public onAddFileClick(): void {
+    (this.input.nativeElement as HTMLInputElement).click();
+  }
+
+  public onFileSelected(event: InputEvent): void {
+    const path = (event.target as HTMLInputElement).files.item(0).path;
+    this.settings.bookmarks.push({
+      url: `file:///${path}`,
```

Comment:
You might want to look into Linux-urls to see if tripple slash works over there too (or if two slashes is enough, and three wouldn't be recognized/supported)

Reply:
Bubblegnash confirmed this should be fine on linux